artichoke-backend/src/extn/core/time/trampoline.rs

```diff
@@ -471,11 +475,49 @@ pub fn subsec(interp: &mut Artichoke, time: Value) -> Result<Value, Error> {
 }
 
 // Time format
+fn strftime_with_encoding(
+    interp: &mut Artichoke,
+    mut time: Value,
+    format: &[u8],
+    encoding: Encoding,
+) -> Result<Value, Error> {
+    let time = unsafe { Time::unbox_from_value(&mut time, interp)? };
+
+    let bytes: Vec<u8> = time.strftime(format).map_err(|e| {
+        // InvalidFormatString is the only true ArgumentError, where as the
+        // rest that can be thrown from strftime are runtime failures.
+        //
+        // ```console
+        // [3.1.2]> Time.now.strftime("%")
+        // (irb):1:in `strftime': invalid format: % (ArgumentError)
+        //      from (irb):1:in `<main>'
+        //	    from /home/ben/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
+        //	    from /home/ben/.rbenv/versions/3.1.2/bin/irb:25:in `load'
+        //	    from /home/ben/.rbenv/versions/3.1.2/bin/irb:25:in `<main>'
+        // ```
+        //
+        // Note: The errors which are re-thrown as RuntimeError include (but is
+        // not limited to: `InvalidTime`, `FormattedStringTooLarge`,
+        // `WriteZero`, `FmtError(Error)`, `OutOfMemory(TryReserveError)`
+        match e {
+            InvalidFormatString => {
+                let mut message = br#"invalid format: "#.to_vec();
+                message.extend_from_slice(format);
+                Error::from(ArgumentError::from(message))
+            }
```

Comment:
`FormattedStringTooLarge` is a dedicated error:

```
[3.1.2] > Time.now.strftime "%4718593m"
(irb):28:in `strftime': Result too large - %4718593m (Errno::ERANGE)
        from (irb):28:in `<main>'
        from /usr/local/var/rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `<main>'
```

We don't have `Errno::Range`, but can we set the right message on an `ArgumentError` and add a TODO to fix the exception class?

Reply:
`WriteZero` also shares this failure mode:

https://github.com/artichoke/strftime-ruby/blob/ccc82452080efa815e31dfe80149e16fd847e9e3/src/tests/format.rs#L875-L876

```
(irb):29:in `strftime': Result too large - %2147483647m (Errno::ERANGE)
        from (irb):29:in `<main>'
        from /usr/local/var/rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `<main>'
```